clientlib/src/DiscreetORM.ts

```diff
@@ -236,25 +239,26 @@ export class StoredClass implements ObjectListener<any>{
     createNewTable(obj: any) : void {
         let table_name = obj.constructor.name;
         let keys = Object.keys(obj);
-        let qmark_arr = new Array<String>(keys.length);
-        qmark_arr.fill('?? ?');
-        let qmark_str = qmark_arr.join(',');
-
-        let create_table_template = `CREATE TABLE ?? (orm_id INT(255), ${qmark_str});`;
-
         // create an array of column name-type strings
-        let args = new Array(keys.length * 2 + 1);
+        let args = new Array(1);
         args[0] = table_name;
+        //used to keep track of the actual 'real' attributes 
+        var count = 0;
         for (let i = 0; i < keys.length; i++) {
-            if (keys[i] === "discreet_orm_id"){
-                // We want to ignore the secreet discreet_orm_id, since discreet_orm_id is already hardcoded in.
-                continue;
+            if(typeof(obj[keys[i]]) != "function" && typeof(obj[keys[i]]) != "undefined" && typeof(obj[keys[i]]) != "object") {
+                // We want to ignore the secret discreet_orm_id, since discreet_orm_id is already hardcoded in.
+                if (keys[i] != "discreet_orm_id"){
+                    args.push(keys[i])
+                    args.push(this.tsTypeToSQLType(obj[keys[i]].constructor.name))
+                    count++;
+                }
             }
-            args[i*2 + 1] = keys[i];
-            args[i*2 + 2] = this.tsTypeToSQLType(obj[keys[i]].constructor.name);
         }
-
         // escape all the user-generated column name strings
+        let qmark_arr = new Array<String>(count);
+        qmark_arr.fill('?? ?');
+        let qmark_str = qmark_arr.join(',');
+        let create_table_template = `CREATE TABLE ?? (orm_id INT(255), ${qmark_str});`;
```

Comment:
Changed the order of execution of this code, since it is not necessary that total number of objects attributes is equal to the total columns in the DB.

Reply:
looks correct to me